src/differentials.jl

```diff
@@ -45,12 +45,12 @@ each step handles an element of the algebra before dispatching to the next step.
 This way, we don't need to implement promotion/conversion rules between subtypes
 of `AbstractDifferential` to resolve potential ambiguities.
 =#
-
+#==
 const PRECEDENCE_LIST = [:wirtinger, :casted, :zero, :dne, :one, :thunk, :fallback]
 
 global defs = Expr(:block)
 
-let previous_add_name = :add, previous_mul_name = :mul
+let previous_add_name = :(Base.:+), previous_mul_name = :(Base.:*)
```

Comment:
This block here could likely be repurposed to define the method hierarchy that resolves ambiguities

Reply:
Yes